### PR TITLE
Add aten::_fused_adam_, aten::_fused_adamw_

### PR DESCRIPTION
**Pull request by hjhee**
- \_fused_adam_
- \_fused_adam_.tensor_lr
- \_fused_adamw_
- \_fused_adamw_.tensor_lr

### CODE REVIEW

**Comment by xytintel** [comment]
Please paste the test cases in the comments.

**Comment by hjhee** [comment]
- <div><b>PASSED</b></div>
<div>test_torch_xpu.py::TestTorchDeviceTypeXPU::test_grad_scaling_autocast_foreach_True_<b>fused1_AdamW</b>_xpu_float32</div>
<div>test_torch_xpu.py::TestTorchDeviceTypeXPU::test_grad_scaling_autocast_foreach_True_<b>fused1_Adam</b>_xpu_float32</div>

**Review by airMeng — COMMENTED** [review]
Only 1k LOC but split into 11 new files, is it the recommended way?

**Comment by xytintel** [comment]
> Only 1k LOC but split into 11 new files, is it the recommended way?

We are currently aligning with CUDA's file organization structure, and a more suitable organization will be discussed later.

File: src/ATen/native/xpu/sycl/fused_adam_impl.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    const double weight_decay,
+    const double eps,
+    const bool maximize,
+    const c10::optional<at::Tensor>& grad_scale,
+    const c10::optional<at::Tensor>& found_inf) {
+  std::vector<std::vector<at::Tensor>> tensor_lists{
+      params.vec(), grads.vec(), exp_avgs.vec(), exp_avg_sqs.vec()};
+
+  if (grad_scale.has_value())
+    TORCH_WARN(" grad_scale not supported with ipex fused_adam.")
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/fused_adam_impl.cpp:28]: Why we have such limitation ?

Comment (hjhee) [reply src/ATen/native/xpu/sycl/fused_adam_impl.cpp]: optional parameters (grad_scale, has_inf), are now supported and aligned to CUDA implementation

**Review by majing921201 — APPROVED** [review]

**Review by xytintel — APPROVED** [review]

**Pull request merged**